### PR TITLE
swapping from and to

### PR DESCRIPTION
**Pull request by jiyoontbd**
Swapping from and to when constructing message, in order to facilitate a conversation thread with correct from/to fields

### CODE REVIEW

**Review by corcillo — APPROVED** [review]

File: js/src/builders.ts [hunk 1]
```diff
@@ -25,8 +25,8 @@ export function createMessage<T extends keyof MessageTypes, U extends keyof Mess
     id,
     threadId,
     parentId,
-    to          : opts.to ?? opts.last.to,
-    from        : opts.from ?? opts.last.from,
+    to          : opts.from ?? opts.last.from,
```
Comment (phoebe-lew) [inline js/src/builders.ts:28]: This reads pretty confusingly to someone without context, maybe add a comment explaining why they need to be swapped?

File: js/src/builders.ts [hunk 2]
```diff
@@ -10,12 +10,23 @@ type LastMessageOpts<T extends keyof MessageTypes> = {
...
   type: T,
   body: MessageType<T>,
 }
 
-export function createMessage<T extends keyof MessageTypes, U extends keyof MessageTypes>(opts: CreateMessageOpts<T, U>): TbDEXMessage<T> {
+export type CreateMessageOpts2<T extends keyof MessageTypes, U extends keyof MessageTypes, X extends keyof MessageTypes> =
+XOR<NewThreadOpts, ReplyMessageOpts<U>> & {
+  type: T,
+  body: MessageType<T>,
+  last?: TbDEXMessage<X>
```
Comment (phoebe-lew) [inline js/src/builders.ts:26]: In which situations do you pass in `last` vs `reply`? And in which situations would you pass in _both_?

File: js/tests/builders.spec.ts [hunk 3]
```diff
@@ -74,8 +74,8 @@ describe('messages builders', () => {
       body : quote
     })
 
-    expect(from).to.equal(rfqMessage.from)
-    expect(to).to.equal(rfqMessage.to)
+    expect(from).to.equal(rfqMessage.to)
```
Comment (phoebe-lew) [inline js/tests/builders.spec.ts:77]: would this make more sense if we renamed to `sender` and `recipient`? then the logic would read like:

`expect(quoteSender).to.equal(rfqMessage.recipient)` which seems somewhat more english-y

File: js/src/builders.ts (hunk 2 above)
Comment (jiyoontbd) [reply js/src/builders.ts:91]: ok we have a smol matrix here

no last, no reply: initial message (i.e. RFQ) - i expect you to pass in to and from

yes last, no reply: replying to a thread where last message is from you (i.e. OrderStatus, Close?) - i expect you to pass in to and from

no last, yes reply:... i can't think of an example here.

yes last, yes reply: replying to a thread where last message is NOT from you (i.e. PFI sends Quote in response to having received RFQ) - i'll fill in the to and from for you

File: js/tests/builders.spec.ts (hunk 3 above)
Comment (jiyoontbd) [reply js/tests/builders.spec.ts]: oh, i like that ^_^

**Comment by jiyoontbd** [comment]
Yeah just close it for now please 🙏

**Pull request closed without merging**